TM1py/Services/GitService.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+import json
+from typing import List
+
+from TM1py.Services.ObjectService import ObjectService
+from TM1py.Services.RestService import RestService, Response
+from TM1py.Objects.GitRemote import GitRemote
+from TM1py.Objects.Git import Git
+from TM1py.Objects.GitCommit import GitCommit
+from TM1py.Objects.GitPlan import GitPushPlan, GitPullPlan, GitPlan
+from TM1py.Utils.Utils import format_url
+
+
+class GitService(ObjectService):
+    """ Service to interact with GIT
+    """
+    def __init__(self, rest: RestService):
+        super().__init__(rest)
+
+    def from_json(self, json_response: str) -> Git:
+        deployed_commit = GitCommit(
+            commit_id=json_response["DeployedCommit"].get("ID"),
+            summary=json_response["DeployedCommit"].get("Summary"),
+            author=json_response["DeployedCommit"].get("Author")
+            )
+
+        remote = GitRemote(
+            connected=json_response["Remote"].get("Connected"),
+            branches=json_response["Remote"].get("Branches"),
+            tags=json_response["Remote"].get("Tags"),
+            )
+
+        git = Git(
+            url=json_response["URL"],
+            deployment=json_response["Deployment"],
+            force=json_response["Deployment"],
+            deployed_commit=deployed_commit,
+            remote=remote)
+
+        return git
+
+    def git_init(self, git_url: str, deployment: str, username: str = None, password: str = None, public_key: str = None,
+                private_key: str = None, passphrase: str = None, force: bool = None, config: dict = None,  **kwargs) -> Git:
+        """ Initialize GIT service, returns Git object
+        :param git_url: file or http(s) path to GIT repository
+        :param deployment: name of selected deployment group
+        :param username: GIT username
+        :param password: GIT password
+        :param public_key: SSH public key, available from PAA V2.0.9.4
+        :param private_key: SSH private key, available from PAA V2.0.9.4
+        :param passphrase: Passphrase for decrypting private key, if set
+        :param force: reset git context on True
+        :param config: Dictionary containing git configuration parameters
+        """
+        url = "/api/v1/GitInit"
+        body = {}
+        body['URL'] = git_url
+        body['Deployment'] = deployment
+
+        if username is not None:
+            body['Username'] = username
+        if password is not None:
+            body['Password'] = password
+        if public_key is not None:
+            body['PublicKey'] = public_key
+        if private_key is not None:
+            body['PrivateKey'] = private_key
+        if passphrase is not None:
+            body['Passphrase'] = passphrase
+        if force is not None:
+            body['Force'] = force
+        if config is not None:
+            body['Config'] = config
+        
+        for key, value in kwargs.items():
```

Comment:
This is not necessary. If you just pass the kwargs through to the POST method the timeout and other technical arguments will be respected.

``` python
self._rest.POST(url=url, data=body, **kwargs)
```